src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOADER_SIZE = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V> [] table;
+
+    MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOADER_SIZE);
+        capacity = DEFAULT_INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = countHash(key);
+        putNode(key, value, index);
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int tableIndex = countHash(key);
+        Node<K, V> currentNode = table[tableIndex];
+        if (currentNode != null) {
+            while (currentNode.next != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            }
+            return currentNode.value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putNode(K key, V value, int index) {
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K, V> currentNode = table[index];
+            do {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    break;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(key, value, null);
+                    size++;
+                }
+                currentNode = currentNode.next;
+            } while (true);
+        }
+    }
+
+    private int countHash(Object key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode()) % capacity;
+    }
```

Comment:
```suggestion
    private int getBucketIndex(K key) {
        return (key == null) ? 0 : Math.abs(key.hashCode()) % capacity;
    }
```
use generic types
you are not calculation hash, you are calculating bucket index

hash and bucket index are not the same thing